zero/gk_geometry.c

```diff
@@ -108,6 +108,30 @@ void gkyl_gk_geometry_augment_local(const struct gkyl_range *inrange,
   }
 }
 
+void gkyl_gk_geometry_bmag_mid(struct gk_geometry* up) {
+  int cdim = up->grid.ndim;
+  int idx_mid[cdim];
+  double xc[cdim];
+  for(int i = 0; i <cdim; i++) {
+    if (up->grid.cells[i]%2 == 0) {
+      idx_mid[i] = up->grid.cells[i]/2;
+      xc[i] = 1.0;
+    }
+    else {
+      idx_mid[i] = up->grid.cells[i]/2+1 ;
```

Comment:
@akashukla better (recall integer division):
```
idx_mid[i] = up->grid.cells[i]/2;
xc[i] = up->grid.cells[i]%2 == 0? 1.0 : 0.0;
```

Reply:
I don't think the first line works the way I want. If up->grid.cells[i] = 4, I want idx_mid[i] = 2. If up->grid.cells[i] = 3, I want 2. But 3/2 = 1 in c right?

Reply:
I'll do this:

Reply:
you can do what I suggested, and evaluate at -1 instead of 1 in the even case, right?

Reply:
I don't think so, because the issue is with the odd case not the even. It is only one extra line now.